packages/browser/src/Component/Widget/Server/SPARQL/View/SparqlQueryWindow.js

```diff
@@ -66,45 +76,58 @@ export class SparqlQueryWindow extends Window {
     this.table = new Table(this);
 
     /**
-     * The initial SPARQL query to display upon window initialization.
+     * Store the queries of the SparqlQueryWindow from the config.
      *
-     * @type {string}
+     * @type {object}
      */
-    this.default_query = `PREFIX rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
-PREFIX owl:  <http://www.w3.org/2002/07/owl#>
-PREFIX xsd:  <http://www.w3.org/2001/XMLSchema#>
-PREFIX gmlowl:  <http://www.opengis.net/ont/gml#>
-PREFIX units: <http://www.opengis.net/def/uom/OGC/1.0/>
-PREFIX geo: <http://www.opengis.net/ont/geosparql#>
-PREFIX geof: <http://www.opengis.net/def/function/geosparql/>
-PREFIX core: <http://www.opengis.net/citygml/2.0/core#>
-PREFIX bldg: <http://www.opengis.net/citygml/building/2.0/building#>
-
-# Return all CityGML City Objects
-SELECT *
-WHERE {
-  ?subject a core:CityModel ;
-    ?predicate ?object .
-  ?subject a ?subjectType .
-  ?object a bldg:Building .
-  ?object a ?objectType .
-  
-  FILTER(?subjectType != owl:NamedIndividual)
-  FILTER(?objectType != owl:NamedIndividual)
-}
-LIMIT 100`;
+    this.queries = configSparql['queries'];
+
     this.registerEvent(Graph.EVENT_NODE_CLICKED);
+    this.registerEvent(Graph.EVENT_NODE_MOUSEOVER);
+    this.registerEvent(Graph.EVENT_NODE_MOUSEOUT);
     this.registerEvent(Table.EVENT_CELL_CLICKED);
   }
 
   /**
    * Override the windowCreated function. Sets the SparqlEndpointResponseProvider
-   * and graph view. Should be called by a `SparqlModuleView`. Once this is done,
+   * and graph view. Also updates this.queries with the queries declared in the configuration file
+   * Should be called by a `SparqlWidgetView`. Once this is done,
    * the window is actually usable ; service event listerers are set here.
-   *
    */
   windowCreated() {
+    // Get queries from text files and update the this.queries
+    new Promise((resolve, reject) => {
```

Comment:
here you could write something like that:

// remove new Promise
const promises = []
// fill promises array
Promise.all(promises).then(()=>{
       // Once query text is updated, update the query select dropdown
      // and query text area
      this.updateQueryDropdown(this.queries);
      this.updateQueryTextArea(0);
      this.updateResultDropdown(0);
})